packages/vitepress-plugin-og-image-generator/src/index.ts

```diff
@@ -0,0 +1,100 @@
+import { fileURLToPath } from 'node:url'
+import { dirname, relative, resolve, sep } from 'node:path'
+import fs from 'fs-extra'
+import fg from 'fast-glob'
+import { Resvg } from '@resvg/resvg-js'
+import type { Config } from './types/config'
+import type { ArticleTree } from './types/metadata'
+import { removeEmoji } from './utils'
+
+export default async function run(config: Config) {
+  const { sidebar, plainTargetDomain, dist } = config
+  const DIR_ROOT = resolve(dirname(fileURLToPath(import.meta.url)), '..')
+
+  // TODO: Load public files in a better way
+  const ogSvg = fs.readFileSync(resolve(DIR_ROOT, './public/og-template.svg'), 'utf-8')
+
+  let articles: ArticleTree[] = ([] as any).concat(...sidebar.map(series => [...series?.items.map(item => ({ ...item, category: series.text }))] || []))
+  for (let i = 0; i < articles.length; i++) {
+    const items = articles[i].items
+    if (items)
+      articles.push(...items.map(item => ({ ...item, category: articles[i].category })))
+  }
+  articles = articles.filter(item => item.link)
+
+  async function generateSVG(article: ArticleTree, output: string) {
+    const lines = removeEmoji(article.text).trim().replace(/(?![^\n]{1,17}$)([^\n]{1,17})\s/g, '$1\n').split('\n')
```

Comment:
This part is hard to read. Can we use line-by-line style when performing chain calling?

```suggestion
    const lines = removeEmoji(article.text)
        .trim()
        .replace(/(?![^\n]{1,17}$)([^\n]{1,17})\s/g, '$1\n')
        .split('\n')
```

Reply:
And what is 17, should we add a comment?